mathphys/imgproc.py

```diff
@@ -0,0 +1,1282 @@
+        """."""
+        res = ''
+        res += f'sizey           : {self.sizey}'
+        res += f'\nsizex           : {self.sizex}'
+        res += f'\nintensity_min   : {self.intensity_min}'
+        res += f'\nintensity_max   : {self.intensity_max}'
+        res += f'\nintensity_avg   : {self.intensity_sum/self.size}'
+        res += f'\nintensity_sum   : {self.intensity_sum}'
+        res += f'\nsaturation_val  : {self.saturation_threshold}'
+        res += f'\nsaturated       : {self.is_saturated}'
+        return res
+
+    @staticmethod
+    def get_roi(data, roix, roiy):
+        """."""
+        if roiy is None:
+            roiy = [0, data.shape[0]]
+        if roix is None:
+            roix = [0, data.shape[1]]
+        roiy[0] = max(roiy[0], 0)
+        roix[0] = max(roix[0], 0)
+        roiy[1] = min(roiy[1], data.shape[0])
+        roix[1] = min(roix[1], data.shape[1])
+        return roix, roiy
+
+    @staticmethod
+    def project_image(data, axis):
+        axis_ = 1 if axis == 0 else 0
+        image = _np.sum(data, axis=axis_)
+        return image
+
+    # --- private methods ---
+
+    def _update_image(self, data):
+        """."""
+        self._data = _np.asarray(data)
+        if self.saturation_threshold is None:
+            self._is_saturated = False
+        else:
+            self._is_saturated = \
+                _np.any(self.data >= self.saturation_threshold)
+
+
+class Image1D_ROI(Image1D):
+    """1D-Image ROI."""
+
+    def __init__(self, data, roi=None, *args, **kwargs):
+        """."""
+        # benchmark for size=1280
+        #   28.8 µs ± 194 ns per loop
+        #   (mean ± std. dev. of 7 runs, 10000 loops each)
+        self._roi = None
+        self._roi_indcs = None
+        self._roi_proj = None
+        self._roi_center = None
+        self._roi_fwhm = None
+        super().__init__(data=data, *args, **kwargs)
+        self._update_image_roi(roi)
+
+    @property
+    def roi(self):
+        """."""
+        return self._roi
+
+    @roi.setter
+    def roi(self, value):
+        """."""
+        self._update_image_roi(value)
+
+    @property
+    def roi_indcs(self):
+        """Image roi indices."""
+        return self._roi_indcs
+
+    @property
+    def roi_proj(self):
+        """Return image roi projection."""
+        return self._roi_proj
+
+    @property
+    def roi_center(self):
+        """Image roi center position."""
+        return self._roi_center
+
+    @property
+    def roi_fwhm(self):
+        """Image roi fwhm."""
+        return self._roi_fwhm
+
+    def update_roi_with_fwhm(self, fwhm_factor=2.0):
+        """."""
+        roi = Image1D_ROI.calc_roi_with_fwhm(self, fwhm_factor)
+        self.roi = roi  # triggers recalc of center, fwhm
+
+    def imshow(
+            self, fig=None, axis=None, crop = None,
+            color_ellip=None, color_roi=None):
+        """."""
+        color_ellip = None if color_ellip == 'no' else color_ellip or 'tab:red'
```

Comment:
instead of using strings here, why not using booleans? I think it is more natural that way.

Reply:
see added docstring